projects/core/src/lib/components/feature-info/feature-info/feature-info.component.ts

```diff
@@ -62,19 +69,31 @@ export class FeatureInfoComponent implements OnInit, OnDestroy {
     this.store$.dispatch(deregisterTool({ tool: ToolbarComponentEnum.FEATURE_INFO }));
   }
 
-  private handleMapClick(evt: { mapCoordinates: [number, number]; mouseCoordinates: [number, number] }) {
+  private handleMapClick(evt: { mapCoordinates: [number, number]; mouseCoordinates: [number, number]; cesiumFeatureInfo?: FeatureInfo3DModel }) {
     combineLatest([
       this.store$.select(selectVisibleLayersWithAttributes),
       this.store$.select(selectVisibleWMSLayersWithoutAttributes),
+      this.store$.select(select3dTilesLayers),
+      this.store$.select(selectIn3DView),
     ])
       .pipe(
         take(1),
-        filter(([ layers, wmsLayers ]) => {
+        filter(([ layers, wmsLayers, tiles3dLayers, in3DView ]) => {
+          if (in3DView) {
+            return layers.length > 0 || wmsLayers.length > 0 || tiles3dLayers.length > 0;
+          }
           return layers.length > 0 || wmsLayers.length > 0;
         }),
         concatMap(() => {
           return this.featureInfoService.fetchFeatures$(evt.mapCoordinates, evt.mouseCoordinates);
         }),
+        tap(() => {
+          this.store$.select(selectIn3DView).pipe(take(1)).subscribe(in3DView => {
```

Comment:
Why not remove this tap, use the evt.cesiumFeatureInfo in `this.featureInfoService.fetchFeatures$` so we have all getting feature info (either from WMS, JDBC or cesiumFeatureInfo) in a single service.